File: go/etl/etl.go
package etl

import "strings"

func Transform(input map[int][]string ) map[string]int{
	res := map[string]int{}

	for k, v := range input {

		for _, i := range v {
			res[strings.ToLower(i)] = k
		}
	}


	return res
}

File: haskell/etl/src/ETL.hs
module ETL (transform) where

import Data.Map (Map, keys, fromList, toList)
import Data.Char (toLower)

transform :: Map a String -> Map Char a
transform legacyData = fromList $ concatMap (\(k, v) -> [ (toLower(x),k) | x <- v ]) $ toList legacyData 
